File: src/Asm.MooBank.Models/Reports/TagTrendReport.cs
namespace Asm.MooBank.Models.Reports;

public record TagTrendReport : ReportBase
{
    public required int TagId { get; init; }

    public required string TagName { get; init; }

    public required IEnumerable<TrendPoint> Months { get; init; }
    public decimal Average { get; set; }
    public decimal? OffsetAverage { get; set; }
}

File: src/Asm.MooBank.Web.App/src/models/reports/TagTrendReport.ts
import { TagSettings } from "../Tag";
import { BaseReport, reportInterval } from "./BaseReport";
import { TrendPoint } from "./TrendPoint";

export interface TagTrendReport extends BaseReport {
    tagId: number,
    tagName: string,
    months: TrendPoint[],
    average: number,
    offsetAverage: number,
}

export interface TrendReportSettings extends Pick<TagSettings, "applySmoothing"> {
    interval: reportInterval,
}

export const defaultSettings: TrendReportSettings = {
    applySmoothing: false,
    interval: "Monthly",
}